yarn-project/simulator/src/avm/opcodes/memory.ts

```diff
@@ -59,16 +59,19 @@ export class Set extends Instruction {
     private value: bigint | number,
   ) {
     super();
+    TaggedMemory.checkIsValidTag(inTag);
```

Comment:
Hmm I see. Ideally we'd do this at deserialization time, but there we don't know if something is a tag so we cant. I guess it's fine to put this here :)